### PR TITLE
photonlibpy: Fix some type check failures

### PR DESCRIPTION
**Pull request by auscompgeek**
This fixes a variety of type check failures raised by both mypy and pyright.

I am however unable to make both completely happy here. In particular an entire method doesn't exist, and I don't know what its implementation is supposed to be. The bulk of mypy's complaints stem from the confusion with names of values that are shadowing modules in the `photonlibpy.generated` package.

```console
> npx pyright photonlibpy
/Users/davidv/dev/frc/photonvision/photon-lib/py/photonlibpy/targeting/multiTargetPNPResult.py
  /Users/davidv/dev/frc/photonvision/photon-lib/py/photonlibpy/targeting/multiTargetPNPResult.py:32:28 - error: Cannot access attribute "createFromPacket" for class "PnpResult"
    Attribute "createFromPacket" is unknown (reportAttributeAccessIssue)
1 error, 0 warnings, 0 informations

> mypy photonlibpy
photonlibpy/targeting/multiTargetPNPResult.py:20: error: Module "photonlibpy.generated.PnpResultSerde" is not valid as a type  [valid-type]
photonlibpy/targeting/multiTargetPNPResult.py:20: note: Perhaps you meant to use a protocol matching the module structure?
photonlibpy/targeting/multiTargetPNPResult.py:32: error: "PnpResult" has no attribute "createFromPacket"  [attr-defined]
photonlibpy/targeting/multiTargetPNPResult.py:40: error: Module "photonlibpy.generated.MultiTargetPNPResultSerde" is not valid as a type  [valid-type]
photonlibpy/targeting/multiTargetPNPResult.py:40: note: Perhaps you meant to use a protocol matching the module structure?
photonlibpy/targeting/TargetCorner.py:13: error: Module "photonlibpy.generated.TargetCornerSerde" is not valid as a type  [valid-type]
photonlibpy/targeting/TargetCorner.py:13: note: Perhaps you meant to use a protocol matching the module structure?
photonlibpy/targeting/photonTrackedTarget.py:66: error: Module "photonlibpy.generated.PhotonTrackedTargetSerde" is not valid as a type  [valid-type]
photonlibpy/targeting/photonTrackedTarget.py:66: note: Perhaps you meant to use a protocol matching the module structure?
photonlibpy/targeting/photonPipelineResult.py:23: error: Module "photonlibpy.generated.PhotonPipelineMetadataSerde" is not valid as a type  [valid-type]
photonlibpy/targeting/photonPipelineResult.py:23: note: Perhaps you meant to use a protocol matching the module structure?
photonlibpy/targeting/photonPipelineResult.py:72: error: Module "photonlibpy.generated.PhotonPipelineResultSerde" is not valid as a type  [valid-type]
photonlibpy/targeting/photonPipelineResult.py:72: note: Perhaps you meant to use a protocol matching the module structure?
photonlibpy/generated/PhotonPipelineResultSerde.py:38: error: generated.PhotonPipelineMetadataSerde? has no attribute "pack"  [attr-defined]
photonlibpy/generated/PhotonPipelineResultSerde.py:53: error: generated.PhotonPipelineMetadataSerde? has no attribute "unpack"  [attr-defined]
photonlibpy/generated/MultiTargetPNPResultSerde.py:37: error: generated.PnpResultSerde? has no attribute "pack"  [attr-defined]
photonlibpy/generated/MultiTargetPNPResultSerde.py:48: error: generated.PnpResultSerde? has no attribute "unpack"  [attr-defined]
photonlibpy/photonCamera.py:61: error: generated.PhotonPipelineResultSerde? has no attribute "MESSAGE_VERSION"  [attr-defined]
photonlibpy/photonCamera.py:130: error: generated.PhotonPipelineResultSerde? has no attribute "unpack"  [attr-defined]
photonlibpy/photonCamera.py:149: error: generated.PhotonPipelineResultSerde? has no attribute "unpack"  [attr-defined]
photonlibpy/photonCamera.py:235: error: generated.PhotonPipelineResultSerde? has no attribute "MESSAGE_VERSION"  [attr-defined]
Found 15 errors in 7 files (checked 18 source files)
```

### CODE REVIEW

File: photon-lib/py/photonlibpy/photonCamera.py [hunk 1]
```diff
@@ -236,12 +236,13 @@ def _versionCheck(self) -> None:
 
         remoteUUID = self._rawBytesEntry.getTopic().getProperty("message_uuid")
 
-        if remoteUUID is None or len(remoteUUID) == 0:
+        if not remoteUUID:
```
Comment (mcm001) [inline photon-lib/py/photonlibpy/photonCamera.py:239]: Needs to also check if renoteUUID is the string "null" and bail if so

```suggestion
        if not remoteUUID or remoteUUID == "null":
```

Comment (auscompgeek) [reply photon-lib/py/photonlibpy/photonCamera.py:239]: Isn't that already covered by the condition below?

Comment (mcm001) [reply photon-lib/py/photonlibpy/photonCamera.py:239]: No. If the remote uuid is "null", that means the coprocessor hasn't published to the topic yet and the property isn't set yet.

Comment (mcm001) [reply photon-lib/py/photonlibpy/photonCamera.py:239]: Anyways; technically out of scope for this PR, but absolutely a bug, and needs to be fixed. seems to also be a bug in c++ as well, potentially:

https://github.com/PhotonVision/photonvision/blob/1fc93bd05dbdb79efc938a4aadf57d49067b4231/photon-targeting/src/main/java/org/photonvision/common/networktables/PacketSubscriber.java#L96

**Review by mcm001 — APPROVED** [review]
Looks to do what it says on the tin :)

**Pull request merged**